server/jetstream_cluster.go

```diff
@@ -7722,17 +7748,38 @@ func (s *Server) jsClusteredConsumerRequest(ci *ClientInfo, acc *Account, subjec
 	}
 }
 
+// caPool pools consumer assignment structs for encoding.
+var caPool = sync.Pool{
+	New: func() any {
+		return &consumerAssignment{}
+	},
+}
+
+func caPoolPut(ca *consumerAssignment) {
+	ciPool.Put(ca.Client)
+	ca.Client = nil
```

Comment:
Same here, do we know js write lock is held? Maybe add comment that the lock must be held?